src/main/javascript/monet.d.ts

```diff
@@ -298,14 +298,26 @@ declare namespace monet {
      * Reader
      */
     
-    interface Reader<E, A> {
-        map<B>(fn: (A) => B): Reader<E, B>;
-        flatMap<B>(fn: (A) => Reader<E, B>): Reader<E, B>;
+    interface Reader<E, A> extends Monad<A> {
+        /* Inherited from Monad: */
+        bind<B>(fn: (val: A) => Reader<E, B>): Reader<E, B>;
+        flatMap<B>(fn: (val: A) => Reader<E, B>): Reader<E, B>;
+        chain<B>(fn: (val: A) => Reader<E, B>): Reader<E, B>;
+        map<B>(fn: (val: A) => B): Reader<E, B>;
+        join<B>(): Reader<E, B>; // if A is Reader<B>
+        takeLeft<X>(m: Reader<E, X>): Reader<E, A>;
+        takeRight<B>(m: Reader<E, B>): Reader<E, B>;
+        ap<B>(rfn: Reader<E, (val: A) => B>): Reader<E, B>;
+
+        /* Reader-specific: */
         run(config: E): A;
+        local<X>(fn: (val: X) => E): Reader<X, A>;
```

Comment:
`init` is plain harmful, so I'm against exposing it.
The other ones are constructors - they are typed on `ReaderStatic`.